core/types/calls.cc

```diff
@@ -1999,14 +1999,13 @@ const vector<Intrinsic> intrinsicMethods{
 
     {Symbols::Class(), Intrinsic::Kind::Instance, Names::new_(), &Class_new},
 
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::buildHash(), &Magic_buildHash},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::buildArray(), &Magic_buildArray},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::expandSplat(), &Magic_expandSplat},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithSplat(), &Magic_callWithSplat},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithBlock(), &Magic_callWithBlock},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithSplatAndBlock(),
-     &Magic_callWithSplatAndBlock},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::suggestType(), &Magic_suggestUntypedConstantType},
+    {Symbols::Magic(), Intrinsic::Kind::Singleton, Names::buildHash(), &Magic_buildHash},
```

Comment:
What's the modification behind the change? Why do you want to call this on an instance of `Magic`?

Reply:
It's just easier to read. The `Magic.<foo>` methods are like the `T.foo` methods.

But for the `Magic` methods, we manually got our hands on the `MagicSingleton` class, and then register the intrinsics as instance methods. but that's exactly what `Intrinsic::Kind::Singleton` does, and was what we were already using the for the `T` methods.

Reply:
Ack, makes sense!